Include linter and fix linter issues

**Problem description**
According to the commonalities v0.3 we need to include the Megalinter to the project and fix linter errors.

**Possible evolution**
Include and configure linter

The results of the linting in the local environment are as follows:

```
sim_swap.yaml
  67:10  warning  camara-operation-summary        Operation Summary Warning: Each operation should include a short summary for better understanding.  paths./retrieve-date.post
  72:11  warning  operation-tag-defined           Operation tags must be defined in global tags.
                              paths./retrieve-date.post.tags[0]
 112:10  warning  camara-operation-summary        Operation Summary Warning: Each operation should include a short summary for better understanding.  paths./check.post
  117:9  warning  operation-tag-defined           Operation tags must be defined in global tags.
                              paths./check.post.tags[0]
 158:12  warning  camara-properties-descriptions  Property description is missing or empty: "openId.description" property must be truthy              components.securitySchemes.openId
 174:17  warning  camara-properties-descriptions  Property description is missing or empty: "SimSwapInfo.description" property must be truthy         components.schemas.SimSwapInfo
 185:22  warning  camara-properties-descriptions  Property description is missing or empty: "CheckSimSwapInfo.description" property must be truthy    components.schemas.CheckSimSwapInfo
 199:15  warning  camara-properties-descriptions  Property description is missing or empty: "ErrorInfo.description" property must be truthy           components.schemas.ErrorInfo
 215:24  warning  camara-properties-descriptions  Property description is missing or empty: "CreateCheckSimSwap.description" property must be truthy  components.schemas.CreateCheckSimSwap
 229:23  warning  camara-properties-descriptions  Property description is missing or empty: "CreateSimSwapDate.description" property must be truthy   components.schemas.CreateSimSwapDate

sim-swap-notification-subscription.yaml
 281:12  warning  camara-properties-descriptions  Property description is missing or empty: "openId.description" property must be truthy                  components.securitySchemes.openId
 310:15  warning  camara-properties-descriptions  Property description is missing or empty: "ErrorInfo.description" property must be truthy               components.schemas.ErrorInfo
 469:20    error  oas3-valid-schema-example       "example" property type must be string
                                  components.schemas.CloudEvent.properties.specversion.example
 490:15    error  oas3-valid-schema-example       "example" property type must be string
                                  components.schemas.Source.example
 504:18  warning  oas3-unused-component           Potentially unused component has been detected.
                                  components.schemas.EventSimSwap
 513:27  warning  oas3-unused-component           Potentially unused component has been detected.
                                  components.schemas.EventSubscriptionEnds
 548:32  warning  camara-properties-descriptions  Property description is missing or empty: "terminationDescription.description" property must be truthy  components.schemas.SubscriptionEnds.properties.terminationDescription
 679:13  warning  camara-properties-descriptions  Property description is missing or empty: "SWAPPED.description" property must be truthy                 components.examples.SWAPPED
 681:13    error  oas3-valid-media-example        "id" property type must be string
                                  components.examples.SWAPPED.value.id
 691:23  warning  camara-properties-descriptions  Property description is missing or empty: "SUBSCRIPTION_ENDS.description" property must be truthy       components.examples.SUBSCRIPTION_ENDS
 693:13    error  oas3-valid-media-example        "id" property type must be string
                                  components.examples.SUBSCRIPTION_ENDS.value.id

✖ 21 problems (4 errors, 17 warnings, 0 infos, 0 hints)
```

For yamllint:
```
sim_swap.yaml
  6:1       error    trailing spaces  (trailing-spaces)
  13:172    error    trailing spaces  (trailing-spaces)
  19:1      error    trailing spaces  (trailing-spaces)
  26:1      error    trailing spaces  (trailing-spaces)
  70:11     error    wrong indentation: expected 12 but found 10  (indentation)
  80:16     error    too many spaces before colon  (colons)
  115:11    error    wrong indentation: expected 12 but found 10  (indentation)
  117:7     error    wrong indentation: expected 8 but found 6  (indentation)
  125:16    error    too many spaces before colon  (colons)
  179:25    error    trailing spaces  (trailing-spaces)

sim-swap-notification-subscription.yaml
  16:24     error    trailing spaces  (trailing-spaces)
  24:82     error    trailing spaces  (trailing-spaces)
  29:53     error    trailing spaces  (trailing-spaces)
```

I propose to clean up the files and then follow the https://github.com/camaraproject/Commonalities/blob/main/documentation/API-linting-Implementation-Guideline.md to implement Github Action in the repository.

--- a/code/API_definitions/sim-swap-notification-subscription.yaml
+++ b/code/API_definitions/sim-swap-notification-subscription.yaml
@@ -13,7 +13,7 @@ info:
     * **SIM swap**:
     A SIM swap is a process in which a mobile phone user's current SIM card is deactivated or replaced with a new one. This is typically done by contacting the user's mobile service provider and requesting a new SIM card for various reasons, such as a lost or damaged SIM card, upgrading to a new phone, or changing phone numbers while keeping the same device.
 
-    # API Functionality    
+    # API Functionality
 
     The API exposes following capabilities:
       * **/subscriptions**: This endpoint has two operations, one that allows to retrieve the list of Sim Swap event subscriptions and another one to create new event subscriptions.
@@ -21,12 +21,12 @@ info:
 
     ## Sim Swap notification subscription
 
-    Theses endpoints allow to manage notification subscription on Sim Swap event. 
+    Theses endpoints allow to manage notification subscription on Sim Swap event.
     The CAMARA subscription model is detailed in the CAMARA API design guideline document and follows Cloudevents specification.
 
     It is mandatory in the subscription to provide the event `type` subscribed are several are managed in this API.
 
-    Only one event ``type`` is managed for this API: 
+    Only one event ``type`` is managed for this API:
       - ``org.camaraproject.sim-swap.v0.swapped`` - Event triggered when a sim swap occurs on the associated phoneNumber
 
 
@@ -466,7 +466,7 @@ components:
         specversion:
           type: string
           description: Version of the specification to which this event conforms (must be 1.0 if it conforms to cloudevents 1.0.2 version)
-          example: 1.0
+          example: "1.0"
         datacontenttype:
           type: string
           description: 'media-type that describes the event payload encoding, must be "application/json" for CAMARA APIs'
@@ -486,14 +486,17 @@ components:
       type: string
       format: uri-reference
       minLength: 1
-      description: "Identifies the context in which an event happened in the specific Provider Implementation."
-      example:
-        - https://github.com/cloudevents
-        - mailto:cncf-wg-serverless@lists.cncf.io
-        - urn:uuid:6e8bc430-9c3a-11d9-9669-0800200c9a66
-        - cloudevents/spec/pull/123
-        - /sensors/tn-1234567/alerts
-        - 1-555-123-4567
+      description: |
+        Identifies the context in which an event happened, as a non-empty `URI-reference` like:
+        - URI with a DNS authority:
+          * https://github.com/cloudevents
+          * mailto:cncf-wg-serverless@lists.cncf.io
+        - Universally-unique URN with a UUID:
+          * urn:uuid:6e8bc430-9c3a-11d9-9669-0800200c9a66
+        - Application-specific identifier:
+          * /cloudevents/spec/pull/123
+          * 1-555-123-4567
+      example: "https://notificationSendServer12.supertelco.com"
 
     DateTime:
       type: string
@@ -678,7 +681,7 @@ components:
   examples:
     SWAPPED:
       value:
-        id: 123655
+        id: "123655"
         source: supertelco.notificationSendServer12
         type: org.camaraproject.sim-swap.v0.swapped
         specversion: "1.0"
@@ -690,7 +693,7 @@ components:
 
     SUBSCRIPTION_ENDS:
       value:
-        id: 123658
+        id: "123658"
         source: supertelco.notificationSendServer12
         type: org.camaraproject.sim-swap.v0.subscription-ends
         specversion: "1.0"

--- a/code/API_definitions/sim_swap.yaml
+++ b/code/API_definitions/sim_swap.yaml
@@ -3,27 +3,27 @@ info:
   title: SIM Swap
   description: |
     The SIM swap API provides a programmable interface for developers and other users (capabilities consumers) to request the last date of a SIM swap performed on the mobile line, or, to check whether a SIM swap has been performed during a past period.
-    
+
     # Introduction
 
     The SIM Swap API performs real-time checks on the last SIM Swap event.
 
     The SIM Swap API is useful to prevent fraud by reducing the risk of account takeover fraud by strengthening SIM based authentication processes such as SMS One-time passwords. Fraudsters are using SIM swap techniques to intercept SMS messages and reset passwords or receive verification codes that allow them to access protected accounts.
 
-    The SIM Swap API can also be used to protect non-automated actions. For example, when a call center expect contacts a user to clarify or confirm a sensitive operation. 
+    The SIM Swap API can also be used to protect non-automated actions. For example, when a call center expect contacts a user to clarify or confirm a sensitive operation.
 
     This API is used by an application to get information about a mobile line latest SIM swap date. It can be easily integrated and used through this secured API and allows SPs (Service Provider) to get this information an easy & secured way. The API provides management of 2 endpoints answering 2 distinct questions:
 
     * When did the last SIM swap occur?
     * Has a SIM swap occurred during last n hours?
-    
+
     Depending on the network provider implementation, legal enforcement, etc... either one or both endpoints could be implemented.
 
     # Relevant terms and definitions
 
     **SIM swap**:
     A SIM swap is a process in which a user's mobile phone number (MSISDN) is associated with a new SIM card (IMSI). This is typically done by contacting the user's mobile service provider and requesting a new SIM card for various reasons, such as a lost or damaged SIM card or upgrading to a new phone.
-    
+
     SimSwap also happens during other actions like changing user's phone number, changing mobile service provider keeping user's mobile phone number or when activating a new SIM associated to the same phone number, known as multisim service.
 
     # API functionality
@@ -67,7 +67,7 @@ paths:
     post:
       security:
         - openId:
-          - sim-swap:retrieve-date
+            - sim-swap:retrieve-date
       tags:
         - Retrieve SIM swap date
       description: Get timestamp of last MSISDN <-> IMSI pairing change for a mobile user account provided with MSIDN.
@@ -77,7 +77,7 @@ paths:
       requestBody:
         description: |
           Create a SIM swap date request for a MSISDN identifier.
-        content :
+        content:
           application/json:
             schema:
               $ref: "#/components/schemas/CreateSimSwapDate"
@@ -112,9 +112,9 @@ paths:
     post:
       security:
         - openId:
-          - sim-swap:check
+            - sim-swap:check
       tags:
-      - Check SIM swap
+        - Check SIM swap
       description: Check if SIM swap has been performed during a past period
       operationId: checkSimSwap
       parameters:
@@ -122,7 +122,7 @@ paths:
       requestBody:
         description: |
           Create a check SIM swap request for a MSISDN identifier.
-        content :
+        content:
           application/json:
             schema:
               $ref: "#/components/schemas/CreateCheckSimSwap"
@@ -176,7 +176,7 @@ components:
       required:
         - latestSimChange
       properties:
-        latestSimChange: 
+        latestSimChange:
           type: string
           format: date-time
           description: Timestamp of latest SIM swap performed. It must follow [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6) and must have time zone. Recommended format is yyyy-MM-dd'T'HH:mm:ss.SSSZ (i.e. which allows 2023-07-03T14:27:08.312+02:00 or 2023-07-03T12:27:08.312Z)

